qfieldsync/core/cloud_api.py

```diff
@@ -688,6 +688,11 @@ def _on_avatar_download_finished(self, reply: QNetworkReply, filename: str) -> N
             self.avatar_success.emit()
 
     def get_last_login_error(self) -> str:
+        try:
+            requests.get(self.url, timeout=5)
+        except requests.ConnectionError:
+            return self.tr("No internet connection. Check your connection.")
```

Comment:
Can you instead use https://doc.qt.io/qt-5/qnetworkreply.html#NetworkError-enum to detect lack of internet?

Reply:
@suricactus I tried with the following but don't return the correct string

```python
request = QNetworkRequest(QUrl(self.url))

reply = self._nam.get(request)

if reply.error() == QNetworkReply.TemporaryNetworkFailureError:
    return self.tr("No internet connection. Check your connection.")
```

![image](https://github.com/user-attachments/assets/63f264d1-e5a3-4e6d-9052-cdf31a1d5e6c)

Reply:
You don't have to send a new request, you already have this information. Check my commit.